xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/VideoFilterShader.h

```diff
@@ -57,11 +57,41 @@ namespace Shaders {
     GLint m_hStretch = 0;
   };
 
+  class BaseVideoFilterShaderGLES : public BaseVideoFilterShaderGL
+  {
+  public:
+    BaseVideoFilterShaderGLES();
+    virtual ~BaseVideoFilterShaderGLES() = default;
+    virtual void OnCompiledAndLinked() override;
+    virtual bool OnEnabled() override;
+    virtual GLint GetVertexLoc() { return m_hVertex; }
+    virtual GLint GetcoordLoc() { return m_hcoord; }
+    virtual void SetMatrices(GLfloat *p, GLfloat *m) { m_proj = p; m_model = m; }
+    virtual void SetAlpha(GLfloat alpha)             { m_alpha = alpha; }
+
+  protected:
+    GLint m_hVertex;
+    GLint m_hcoord;
+    GLint m_hProj;
+    GLint m_hModel;
+    GLint m_hAlpha;
+
+    GLfloat *m_proj;
+    GLfloat *m_model;
+    GLfloat  m_alpha;
+  };
+
+#if HAS_GLES >= 2
+  using BaseVideoFilterShader = BaseVideoFilterShaderGLES;
```

Comment:
I don't like this. That's kind of the same hack the fist GLES rendering fo vaapi had shown.